trade-finance-manager-api/src/v1/validation/route-validators/update-report-status-payload-validation/index.test.ts

```diff
@@ -0,0 +1,215 @@
+      {
+        status: opts.status ?? UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION,
+        report: opts.report ?? {
+          month: 5,
+          year: 2023,
+          bankId: '123',
+        },
+      },
+    ],
+  });
+
+  const getUpdateReportStatusPayloadValidationResult = async (req: Request) => {
+    const res = createResponse();
+    const next = jest.fn();
+
+    const validators = updateReportStatusPayloadValidation.map((validator) => {
+      if (Array.isArray(validator)) {
+        return validator.flatMap((subValidator) => subValidator(req, res, next));
+      }
+      return validator(req, res, next);
+    });
+    await Promise.all(validators);
+
+    return validationResult(req).array();
+  };
+
+  describe('for a payload with a bank id, month and year combination', () => {
+    const validPayload = {
+      month: 5,
+      year: 2023,
+      bankId: '123',
+    };
+
+    it.each([
+      // @ts-expect-error month is supposed to be the incorrect type for this test
+      ['month is a string', getValidPayloadBody({ report: { ...validPayload, month: '5' } })],
+      ['month is less than 1', getValidPayloadBody({ report: { ...validPayload, month: 0 } })],
+      ['month is greater than 12', getValidPayloadBody({ report: { ...validPayload, month: 13 } })],
+      // @ts-expect-error year is supposed to be the incorrect type for this test
+      ['year is a string', getValidPayloadBody({ report: { ...validPayload, year: '2023' } })],
+      ['year is less than 0', getValidPayloadBody({ report: { ...validPayload, year: -1 } })],
+      // @ts-expect-error bankId is supposed to be the incorrect type for this test
+      ['bankId is a number', getValidPayloadBody({ report: { ...validPayload, bankId: 123 } })],
+    ])('returns a single error when one of the parameters in invalid (%s)', async (_, payload) => {
+      // Arrange
+      const req = createRequest({ body: payload });
+
+      // Act
+      const errors = await getUpdateReportStatusPayloadValidationResult(req);
+
+      // Assert
+      expect(errors.length).toEqual(1);
+    });
+
+    it('returns no errors when the payload is valid', async () => {
+      // Arrange
+      const body = getValidPayloadBody({ report: validPayload });
+      const req = createRequest({ body });
+
+      // Act
+      const errors = await getUpdateReportStatusPayloadValidationResult(req);
+
+      // Assert
+      expect(errors.length).toEqual(0);
+    });
+  });
+
+  describe('for a payload with a report id', () => {
+    const validMongoId = '5ce819935e539c343f141ece';
+
+    it('returns a single error when the report id is not a valid mongo id', async () => {
+      // Arrange
+      const report = { id: '123' };
+      const body = getValidPayloadBody({ report });
+      const req = createRequest({ body });
+
+      // Act
+      const errors = await getUpdateReportStatusPayloadValidationResult(req);
+
+      // Assert
+      expect(errors.length).toEqual(1);
```

Comment:
I did think about this when I first wrote them, but decided not to (largely out of laziness, but also because extracting the error message was a bit of a pain if the tests failed). I will update these tests though now that they are passing